client/js/components/main/assessment.jsx

```diff
@@ -210,6 +254,34 @@ export class Assessment extends React.Component{
     return warning;
   }
 
+  /**
+   * Returns NextButton if next questions are unlocked,  CheckAnswerButton
+   * otherwise.
+   */
+  getNextButton(){
```

Comment:
Similarly I was having a hard time naming this as well. This contains the logic to determine whether or not the next button or the check answer should be rendered.